test/requests/support/sorbet_test.rb

```diff
@@ -0,0 +1,834 @@
+    refute(annotation?("type_template(x, y)"))
+    refute(annotation?("type_template(x) {}"))
+    refute(annotation?("type_template(x, y) {}"))
+
+    refute(annotation?("T.type_template"))
+    refute(annotation?("T.type_template()"))
+    refute(annotation?("T.type_template(x)"))
+    refute(annotation?("T.type_template(x, y)"))
+    refute(annotation?("T.type_template {}"))
+    refute(annotation?("T.type_template() {}"))
+    refute(annotation?("T.type_template(x) {}"))
+    refute(annotation?("T.type_template(x, y) {}"))
+
+    refute(annotation?("C.type_template"))
+    refute(annotation?("C.type_template()"))
+    refute(annotation?("C.type_template(x)"))
+    refute(annotation?("C.type_template(x, y)"))
+    refute(annotation?("C.type_template {}"))
+    refute(annotation?("C.type_template() {}"))
+    refute(annotation?("C.type_template(x) {}"))
+    refute(annotation?("C.type_template(x, y) {}"))
+  end
+
+  def test_unsafe
+    assert(annotation?("unsafe(x)"))
+
+    refute(annotation?("unsafe"))
+    refute(annotation?("unsafe()"))
+    refute(annotation?("unsafe(x, y)"))
+    refute(annotation?("unsafe {}"))
+    refute(annotation?("unsafe() {}"))
+    refute(annotation?("unsafe(x) {}"))
+    refute(annotation?("unsafe(x, y) {}"))
+
+    refute(annotation?("T.unsafe"))
+    refute(annotation?("T.unsafe()"))
+    refute(annotation?("T.unsafe(x)"))
+    refute(annotation?("T.unsafe(x, y)"))
+    refute(annotation?("T.unsafe {}"))
+    refute(annotation?("T.unsafe() {}"))
+    refute(annotation?("T.unsafe(x) {}"))
+    refute(annotation?("T.unsafe(x, y) {}"))
+
+    refute(annotation?("C.unsafe"))
+    refute(annotation?("C.unsafe()"))
+    refute(annotation?("C.unsafe(x)"))
+    refute(annotation?("C.unsafe(x, y)"))
+    refute(annotation?("C.unsafe {}"))
+    refute(annotation?("C.unsafe() {}"))
+    refute(annotation?("C.unsafe(x) {}"))
+    refute(annotation?("C.unsafe(x, y) {}"))
+  end
+
+  def test_untyped
+    assert(annotation?("T.untyped"))
+    assert(annotation?("T.untyped()"))
+
+    refute(annotation?("T.untyped(x)"))
+    refute(annotation?("T.untyped(x, y)"))
+    refute(annotation?("T.untyped {}"))
+    refute(annotation?("T.untyped() {}"))
+    refute(annotation?("T.untyped(x) {}"))
+    refute(annotation?("T.untyped(x, y) {}"))
+
+    refute(annotation?("untyped"))
+    refute(annotation?("untyped()"))
+    refute(annotation?("untyped(x)"))
+    refute(annotation?("untyped(x, y)"))
+    refute(annotation?("untyped {}"))
+    refute(annotation?("untyped() {}"))
+    refute(annotation?("untyped(x) {}"))
+    refute(annotation?("untyped(x, y) {}"))
+
+    refute(annotation?("C.untyped"))
+    refute(annotation?("C.untyped()"))
+    refute(annotation?("C.untyped(x)"))
+    refute(annotation?("C.untyped(x, y)"))
+    refute(annotation?("C.untyped {}"))
+    refute(annotation?("C.untyped() {}"))
+    refute(annotation?("C.untyped(x) {}"))
+    refute(annotation?("C.untyped(x, y) {}"))
```

Comment:
This seems like a lot of tests due to the combinations, I'm sure they're very fast but wonder if it's necessary to be as exhaustive?